tensorflow/core/kernels/crop_and_resize_op.cc

```diff
@@ -216,7 +217,73 @@ struct CropAndResize<CPUDevice, T> {
     const int depth = crops.dimension(3);
 
     // Sharding across boxes.
-    auto CropAndResizePerBox = [&](int start_box, int limit_box) {
+    std::function<void(int32 b_in, int b, int y, const float x1, const float x2,
```

Comment:
Please avoid storing lambda functions in `std::function`. I don't think `std::function` can be inlined, which will make the whole thing slower. Please use `auto` instead. The same goes for other lambdas in this PR. Thank you!

Reply:
I'm afraid we can't simply use `auto` to replace `std::function`, because a lambda function is const thus can't be assign to a new value in the following 'nearest' case line.

Reply:
I propose following 2 alternative approaches, if you insist to use lambda functions:
1. Change the whole structure of the wrapper function to:  
`if (method_name == "bilinear") {
         auto interpolation_fn = [&](){...}
         auto CropAndResizeBackImagePerBox = [&](){...}
         // A bunch of code
         Shard(worker_threads.num_threads...
         return true;
     else {
         auto interpolation_fn = [&](){...}
         auto CropAndResizeBackImagePerBox = [&](){...}
         // A bunch of code
         Shard(worker_threads.num_threads...
         return true;
     }
 `

Reply:
We can use a ternary operator 
```
auto lambda = (method_name == "bilinear ? method_a : method_b;
```

But I totally agree with your 2. Let's do 2. Thank you! :)

Reply:
2 applied! We can move on now.